apps/backoffice-v2/src/common/components/molecules/DateRangePicker/DateRangePicker.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { useEffect, useState, useRef } from 'react';
+import { CalendarIcon } from '@radix-ui/react-icons';
+import { formatDate } from '@/common/utils/format-date';
+import { DateRange } from 'react-day-picker';
+import { addMonths } from 'date-fns';
+import { ctw } from '@/common/utils/ctw/ctw';
+import { Button } from '../../atoms/Button/Button';
+import { Calendar } from '../../organisms/Calendar/Calendar';
+import { Popover, PopoverContent, PopoverTrigger } from '@ballerine/ui';
+
+type DateRangePickerProps = {
+  onChange: (range: { start: Date | null; end: Date | null }) => void;
+  value: { start: Date | null; end: Date | null };
+};
+
+export function DateRangePicker(
+  { onChange, value }: DateRangePickerProps,
+  { className }: React.HTMLAttributes<HTMLDivElement>,
+) {
+  const currentDate = new Date();
+  const [date, setDate] = useState<DateRange | undefined>({
+    from: new Date(currentDate.getFullYear(), currentDate.getMonth(), currentDate.getDate()),
+    to: addMonths(
+      new Date(currentDate.getFullYear(), currentDate.getMonth(), currentDate.getDate(), 1),
+      1,
+    ),
+  });
+  const prevDateRef = useRef<DateRange | undefined>(date);
+
+  useEffect(() => {
+    if (date !== prevDateRef.current) {
+      onChange({
+        start: date?.from || null,
+        end: date?.to || null,
+      });
+      prevDateRef.current = date;
+    }
+  }, [date, onChange]);
```

Comment:
`useEffect` should be a last resort. Here passing the date from outside as the `useState`'s initial value and using the `onChange` on the component itself would behave in a more robust way and more consistently. Check this if you haven't yet. https://react.dev/learn/you-might-not-need-an-effect